Assets/Scripts/Controllers/AI/Boss/Boss.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class Boss : MonoBehaviour {
+	public BossStatsSO Stats;
+	[SerializeField] public BossAttacks BossAttacks;
```

Comment:
Why are references to components on the boss gameobject serialized? Use getcomponent<>() in the Start or Awake functions please.